library/data/recipe/src/main/java/org/quiltmc/qsl/recipe/api/brewing/PotionBrewingRecipe.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.recipe.api.brewing;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import com.google.gson.JsonObject;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.network.PacketByteBuf;
+import net.minecraft.potion.Potion;
+import net.minecraft.potion.PotionUtil;
+import net.minecraft.recipe.Ingredient;
+import net.minecraft.recipe.RecipeSerializer;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.JsonHelper;
+import net.minecraft.util.registry.Registry;
+
+import org.quiltmc.qsl.recipe.impl.RecipeImpl;
+
+/**
+ * A {@link Potion} brewing recipe.
+ *
+ * <p>
+ * The recipe has six arguments:
+ * <ul>
+ * 		<li>type: "quilt_recipe:potion_brewing"</li>
+ * 		<li>group: A string representing the group of the recipe</li>
+ * 		<li>ingredient: A valid {@link net.minecraft.recipe.Ingredient ingredient} JSON object.</li>
+ * 		<li>input: A valid {@link net.minecraft.util.Identifier identifier} for a {@link Potion potion}.</li>
+ * 		<li>output: A valid {@link net.minecraft.util.Identifier identifier} for a {@link Potion potion}.</li>
+ * 		<li>fuel: An integer representing how much fuel this craft will take.
+ * 			In vanilla, blaze powder supplies {@code 20} fuel.</li>
+ * 		<li>time: An integer representing how much time this craft will take, in ticks.
+ * 			In vanilla, the default is {@code 400} ticks.</li>
+ * </ul>
+ *
+ * Here is an example recipe for a potion of luck that takes a water potion, a trapdoor of some kind, {@code 5} fuel units, and {@code 123} ticks.
+ * <pre><code>
+ * {
+ *   "type": "quilt_recipe:potion_brewing",
+ *   "ingredient": {
+ *     "tag": "minecraft:trapdoors"
+ *   },
+ *   "input": "minecraft:water",
+ *   "output": "minecraft:luck",
+ *   "fuel": 5,
+ *   "time": 123
+ * }
+ * </code></pre>
+ */
+public class PotionBrewingRecipe extends AbstractBrewingRecipe<Potion> {
```

Comment:
Maybe `SimplePotionBrewingRecipe` to help distinguish it with the `Custom` one and the `Abstract` one?